### PR TITLE
Update packages/as-types/argument.ts documentation

### LINKED ISSUE #126: arguments documentation

### PR DESCRIPTION
**Pull request by Elli610**
closes #126

### CODE REVIEW

File: packages/as-types/assembly/argument.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export class Args {
   private serialized: StaticArray<u8> = new StaticArray<u8>(0);
 
   /**
-   * Initializes a new instance of Args.
+   * Initialize a new instance of Args.
```
Comment (Ben-Rey) [inline packages/as-types/assembly/argument.ts:45]: Initializes was correct I think

**Review by Ben-Rey — CHANGES_REQUESTED** [review]

File: packages/as-types/assembly/argument.ts [hunk 2]
```diff
@@ -42,22 +42,26 @@ export class Args {
   private serialized: StaticArray<u8> = new StaticArray<u8>(0);
```
Comment (0xSwapFeeder) [inline packages/as-types/assembly/argument.ts:43]: Missing @remarks in class description ?
Maybe use @link to talk about specific functions from the class also.

File: packages/as-types/assembly/argument.ts [hunk 3]
```diff
@@ -42,22 +42,26 @@ export class Args {
...
 
   /**
+   * Initializes a new instance of Args.
+   *
+   * @param serialized - The serialized arguments to pass to CallSC. Default: []
```
Comment (0xSwapFeeder) [inline packages/as-types/assembly/argument.ts:47]: What is CallSC ?
I mean I might know, but Args is not only made for smart contract calls, this seems a bit out of context.

File: packages/as-types/assembly/argument.ts [hunk 4]
```diff
@@ -42,22 +42,26 @@ export class Args {
...
+   */
   get offset(): i32 {
     return this._offset;
   }
 
   /**
-   * Returns the serialized string to pass to CallSC.
-   *
-   * @returns the serialized string
+   * @returns the serialized argument to pass to CallSC.
```
Comment (0xSwapFeeder) [inline packages/as-types/assembly/argument.ts:64]: Same here with call SC

File: packages/as-types/assembly/argument.ts [hunk 5]
```diff
@@ -232,7 +309,16 @@ export class Args {
   }
 
   /**
-   * Returns the deserialized number as f32.
+   * This method deserializes an Ff2 from a serialized array starting from the current offset.
```
Comment (0xSwapFeeder) [inline packages/as-types/assembly/argument.ts:312]:
```suggestion
   * This method deserializes an F32 from a serialized array starting from the current offset.
```

File: packages/as-types/assembly/argument.ts [hunk 6]
```diff
@@ -325,9 +454,12 @@ export class Args {
   }
 
   /**
-   * Returns the data of requested size for current offset
+   * This function retrieves the next chunk of data from the serialized data array based on the specified size
+   * and returns it as a static array of unsigned 8-bit integers.
```
Comment (0xSwapFeeder) [inline packages/as-types/assembly/argument.ts:458]: Maybe add a @remarks to explain that calling this function (unlike other 'next') doesn't increment the offset and therefore is quite usefull to fetch data directly and still be able to use other functions afterwards.
And also that using this function might produce duplicates of the data fetched for the same reasons.

**Review by 0xSwapFeeder — CHANGES_REQUESTED** [review]

File: packages/as-types/assembly/argument.ts [hunk 7]
```diff
@@ -66,9 +71,17 @@ export class Args {
   // getters
 
   /**
-   * Returns the deserialized string.
+   * This method deserializes a string from a serialized array starting from the current offset.
```
[the remark this answers is not in the document]
Comment (Ben-Rey) [reply packages/as-types/assembly/argument.ts:74]: Same for the others, no need to repeat This method every time

**Review by Ben-Rey — CHANGES_REQUESTED** [review]
Some methods doesn't have title. It's repetitive but I would prefer to keep the style consistent.

File: packages/as-types/assembly/argument.ts [hunk 8]
```diff
@@ -26,38 +26,43 @@ import {
...
+   */
   get offset(): i32 {
     return this._offset;
   }
 
   /**
-   * Returns the serialized string to pass to CallSC.
-   *
-   * @returns the serialized string
+   * @returns the serialized argument.
```
[the remark this answers is not in the document]
Comment (Ben-Rey) [reply packages/as-types/assembly/argument.ts:65]: No tags in title, it will be weird in the generated doc

**Review by 0xSwapFeeder — APPROVED** [review]
@Ben-Rey lgtm

**Review by Ben-Rey — APPROVED** [review]

**Pull request merged**